docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,179 @@
+# Allowing OpenShift Groups to be Synced With Sources on an LDAP Server and Making it Extensible for Other External Sources
```

Comment:
I was envisioning the entire `GroupSyncer` to be an abstraction so the `LDAPGroupSyncer` could just implement that. If it's not useful we don't need it, but I'm not able to make that call due to my lack of any sys-admin experience.